docs/source/hyper-parameter-search.rst

```diff
@@ -73,12 +79,52 @@ to pull it locally to your computer:
     # Pass to fit without ever leaving the cluster
     search.fit(df[['x', 'x2']], df['y'])
 
+.. _hyperband:
+
+State-of-the-art algorithms
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
This section title seems a little grandiose.

Reply:
I think that we should qualify here that Hyperband is only appropriate for incremental learners.  I'm somewhat concerned that this documentation section might oversell the algorithm.  Our goal here is to guide users towards correct algorithms given their situation.  Please be aware that incremental learning may or may not be the common case among users.

Reply:
Ah, I see now that this is in a selling points section of the documentation.  

Instead, I wonder if we might provide guidance to users on which hyperparameter algorithm to choose given their situation?

If you want to write about Hyperband in more detail as well that's good too, but that detail probably shouldn't dominate the main user doc page on the subject.  Maybe add more information to the docstring and then include the class in API documentation with nice summary links using `.. autosummary:: HyperbandCV` ?

Reply:
I've tried to give more of a fair summary of Hyperband here. I mention the limitations on cross-validation in the docstring, and link to #250 in the docstring. Any feedback?